src/data-source/collection-browser-data-source.ts

```diff
@@ -0,0 +1,1189 @@
+        collections: result.collection?.values ?? [],
+        collectionFilesCount: result.collection_files_count?.value ?? 0,
+        collectionSize: result.collection_size?.value ?? 0,
+        commentCount: result.num_reviews?.value ?? 0,
+        creator: result.creator?.value,
+        creators: result.creator?.values ?? [],
+        dateAdded: result.addeddate?.value,
+        dateArchived: result.publicdate?.value,
+        datePublished: result.date?.value,
+        dateReviewed: result.reviewdate?.value,
+        description: result.description?.values.join('\n'),
+        favCount: result.num_favorites?.value ?? 0,
+        href: this.collapseRepeatedQuotes(result.__href__?.value),
+        identifier: result.identifier,
+        issue: result.issue?.value,
+        itemCount: result.item_count?.value ?? 0,
+        mediatype: this.getMediatype(result),
+        snippets: result.highlight?.values ?? [],
+        source: result.source?.value,
+        subjects: result.subject?.values ?? [],
+        title: result.title?.value ?? '',
+        volume: result.volume?.value,
+        viewCount: result.downloads?.value ?? 0,
+        weeklyViewCount: result.week?.value,
+        loginRequired,
+        contentWarning,
+      });
+    });
+    this.addPage(pageNumber, tiles);
+    const visiblePages = this.host.currentVisiblePageNumbers;
+    const needsReload = visiblePages.includes(pageNumber);
+    if (needsReload) {
+      this.host.refreshVisibleResults();
+    }
+  }
+
+  /**
+   * Returns the mediatype string for the given search result, taking into account
+   * the special `favorited_search` hit type.
+   * @param result The search result to extract a mediatype from
+   */
+  private getMediatype(result: SearchResult): MediaType {
+    /**
+     * hit_type == 'favorited_search' is basically a new hit_type
+     * - we are getting from PPS.
+     * - which gives response for fav- collection
+     * - having favorited items like account/collection/item etc..
+     * - as user can also favorite a search result (a search page)
+     * - so we need to have response (having fav- items and fav- search results)
+     *
+     * if backend hit_type == 'favorited_search'
+     * - let's assume a "search" as new mediatype
+     */
+    if (result?.rawMetadata?.hit_type === 'favorited_search') {
+      return 'search';
+    }
+
+    return result.mediatype?.value ?? 'data';
+  }
+
+  /**
+   * Returns the input string, but removing one set of quotes from all instances of
+   * ""clauses wrapped in two sets of quotes"". This assumes the quotes are already
+   * URL-encoded.
+   *
+   * This should be a temporary measure to address the fact that the __href__ field
+   * sometimes acquires extra quotation marks during query rewriting. Once there is a
+   * full Lucene parser in place that handles quoted queries correctly, this can likely
+   * be removed.
+   */
+  private collapseRepeatedQuotes(str?: string): string | undefined {
```

Comment:
suggestion: non instance bound functions, let's move to helpers, util area...

Reply:
Nice catch, I'll move these out of the data source class!